src/features/Library/Header/Header.tsx

```diff
@@ -0,0 +1,87 @@
+import { useNavigate } from 'react-router-dom';
+import { useTranslation } from 'react-i18next';
+import { Box } from '@mui/material';
+
+import { Search, Svg } from 'components';
+import { StyledBodySmall, StyledLabelLarge } from 'styles/styledComponents';
+import theme from 'styles/theme';
+import { variables } from 'styles/variables';
+import { page } from 'resources';
+
+import {
+  StyledHeaderContainer,
+  StyledBackButton,
+  StyledBuilderButton,
+  StyledCartButton,
+} from './Header.styles';
+import { HeaderProps, RightButtonType } from './Header.types';
+
+export const Header = ({
+  isBackButtonVisible = false,
+  handleSearch,
+  rightButtonType,
+  rightButtonCallback,
+}: HeaderProps) => {
+  const { t } = useTranslation('app');
+  const navigate = useNavigate();
+
+  const appletsCount = 0; // TODO: get the number of applets from the cart
+
+  const handleNavigate = () => {
+    navigate(page.library);
+  };
+
+  const renderRightButton = () => {
+    switch (rightButtonType) {
+      case RightButtonType.Cart:
+        return (
+          <StyledCartButton
+            startIcon={<Svg width="18" height="18" id="cart-filled" />}
+            endIcon={<Svg width="20" height="20" id="navigate-right" />}
+            variant="outlined"
+            onClick={rightButtonCallback}
+          >
+            <Box sx={{ marginLeft: theme.spacing(1.2) }}>
+              <StyledLabelLarge sx={{ textAlign: 'initial', color: variables.palette.on_surface }}>
+                {t('cart')}
+              </StyledLabelLarge>
+              <StyledBodySmall sx={{ color: variables.palette.on_surface_variant }}>
+                {appletsCount} {t('applet', { count: appletsCount })}
+              </StyledBodySmall>
+            </Box>
+          </StyledCartButton>
+        );
+      case RightButtonType.Builder:
+        return (
+          <StyledBuilderButton
+            onClick={rightButtonCallback}
+            startIcon={<Svg width="18" height="18" id="builder" />}
+            variant="contained"
+          >
+            {t('addToBuilder')}
+          </StyledBuilderButton>
+        );
+    }
+  };
+
+  return (
+    <StyledHeaderContainer>
+      <Box>
```

Comment:
do you need empty box?

Reply:
there is for grid, because in case when there is no back button, the empty space should be in this place